test/integration/policy_report_metric_test.go

```diff
@@ -0,0 +1,167 @@
+		By("Got the metrics route url: " + routeHost)
+		insightsMetricsURL = "https://" + routeHost + "/metrics"
+
+		//get auth token from service account
+		By("Setting up ServiceAccount for authentication")
+		_, err = common.OcHub("create", "serviceaccount", saName, "-n", userNamespace)
+		Expect(err).To(BeNil())
+		_, err = common.OcHub("create", "clusterrolebinding", roleBindingName, "--clusterrole=cluster-admin", fmt.Sprintf("--serviceaccount=%s:%s", userNamespace, saName))
+		Expect(err).To(BeNil())
+		tokenNames, err := common.OcHub("get", fmt.Sprintf("serviceaccount/%s", saName), "-n", userNamespace, "-o", "jsonpath={.secrets[*].name}")
+		Expect(err).To(BeNil())
+		tokenNameArr := strings.Split(tokenNames, " ")
+		var tokenName string
+		for _, name := range tokenNameArr {
+			if strings.HasPrefix(name, saName+"-token-") {
+				tokenName = name
+			}
+		}
+		encodedtoken, err := common.OcHub("get", "secret", tokenName, "-n", userNamespace, "-o", "jsonpath={.data.token}")
+		Expect(err).To(BeNil())
+		decodedToken, err := base64.StdEncoding.DecodeString(encodedtoken)
+		Expect(err).To(BeNil())
+		insightsToken = string(decodedToken)
+	})
+	It("Checks that the endpoint does not expose metrics without auth", func() {
+		Eventually(func() interface{} {
+			_, status, err := common.GetWithToken(insightsMetricsURL, "")
+			if err != nil {
+				return err
+			}
+			return status
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring("Unauthorized"))
+	})
+	It("Checks that a noncompliant policy reports a metric", func() {
+		By("Creating a noncompliant policy")
+		common.OcHub("apply", "-f", noncompliantPolicyYamlReport, "-n", userNamespace)
+		Eventually(
+			getComplianceState(noncompliantPolicyNameReport),
+			defaultTimeoutSeconds,
+			1,
+		).Should(Equal(policiesv1.NonCompliant))
+
+		By("Checking the policy metric")
+		policyLabel := `policy="` + userNamespace + "." + noncompliantPolicyNameReport + `"`
+		Eventually(func() interface{} {
+			resp, _, err := common.GetWithToken(insightsMetricsURL, strings.TrimSpace(insightsToken))
+			if err != nil {
+				return err
+			}
+			return resp
+		}, 120, 1).Should(common.MatchMetricValue(insightsMetricName, policyLabel, "1"))
+	})
+	It("Checks that changing the policy to compliant removes the metric", func() {
+		By("Creating a compliant policy")
+		common.OcHub("apply", "-f", compliantPolicyYamlReport, "-n", userNamespace)
+		Eventually(
+			getComplianceState(compliantPolicyNameReport),
+			defaultTimeoutSeconds*2,
+			1,
+		).Should(Equal(policiesv1.Compliant))
+
+		By("Checking the policy metric displays nothing")
+		policyLabel := `policy="` + userNamespace + "." + noncompliantPolicyNameReport + `"`
+		Eventually(func() interface{} {
+			resp, _, err := common.GetWithToken(insightsMetricsURL, strings.TrimSpace(insightsToken))
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds*2, 1).Should(common.MatchMetricValue(insightsMetricName, policyLabel, "1"))
```

Comment:
Since we're testing for the *lack* of this metric, should this be a `ShouldNot`?